aws/data_source_aws_network_interface.go

```diff
@@ -132,20 +133,37 @@ func dataSourceAwsNetworkInterface() *schema.Resource {
 func dataSourceAwsNetworkInterfaceRead(d *schema.ResourceData, meta interface{}) error {
 	conn := meta.(*AWSClient).ec2conn
 
-	input := &ec2.DescribeNetworkInterfacesInput{
-		NetworkInterfaceIds: []*string{aws.String(d.Get("id").(string))},
+	filters, filtersOk := d.GetOk("filter")
+	id, idOk := d.GetOk("id")
+	if idOk && filtersOk {
+		return fmt.Errorf("Only one of filter or id can be assigned")
```

Comment:
Is there a technical reason for this limitation?
* If both `id` and `filter` are supplied and the filter successfully matches the resource corresponding to the ID, its okay and the data source will return the one result.
* If both `id` and `filter` are supplied and the filter doesn't successfully match the resource corresponding to the ID, the data source will return the error.
* Folks might want to create a module with this data source and toggle between both attributes. HCL doesn't support a `nil` value at the moment so this would prevent that use case.